### PR TITLE
Disable tests that fail to build under 3.0.0-preview3

### PR DESCRIPTION
**Pull request by jashook**

### CODE REVIEW

File: tests/src/Loader/classloader/generics/Instantiation/Nesting/NestedGenericClasses.csproj [hunk 1]
```diff
@@ -12,6 +12,9 @@
     <OutputType>Exe</OutputType>
     <CLRTestKind>BuildAndRun</CLRTestKind>
     <CLRTestPriority>1</CLRTestPriority>
+
+    <!-- Test fails to build on .Net Core 3.0 preview3 -->
+    <DisableProjectBuild Condition="'$(__BuildOS)' == 'OSX'">true</DisableProjectBuild>
```
Comment (tannergooding) [inline tests/src/Loader/classloader/generics/Instantiation/Nesting/NestedGenericClasses.csproj]: Just a note, these look to also fail on Windows if using the desktop MSBuild.

**Comment by jashook** [comment]
See https://github.com/dotnet/coreclr/issues/22757

File: tests/src/Loader/classloader/generics/Instantiation/Nesting/NestedGenericClasses.csproj (hunk 1 above)
Comment (jashook) [reply tests/src/Loader/classloader/generics/Instantiation/Nesting/NestedGenericClasses.csproj]: We do not have an exclude mechanism for desktop. So I believe this change is ready to merge.

**Comment by jashook** [comment]
Failures are unrelated.

OSX: Timeout
CoreFX: tests are unaffected by this change.
Arm: Test failure https://github.com/dotnet/coreclr/issues/22784
coreclr-ci: Timeout

File: tests/src/Loader/classloader/generics/Instantiation/Nesting/NestedGenericClasses.csproj (hunk 1 above)
Comment (tannergooding) [reply tests/src/Loader/classloader/generics/Instantiation/Nesting/NestedGenericClasses.csproj]: @jashook, you should just be able to test `'$(RunningOnCore)' != 'true'`, which is how you determine if you are running on the Desktop or .NET Core MSBuild (and therefore, which compiler task you will be invoking).

**Pull request merged**